# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../specter/specter-1.4-0001-postgres-plugin-initialization.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description: Fixed postgres plugin initialization
Origin: Upstream (http://bazaar.launchpad.net/~ruby/specter/trunk/diff/2)

=== modified file 'plugins/specter_PGSQL.c'
--- specter-1.4/plugins/specter_PGSQL.c	2008-01-06 17:26:23 +0000
+++ specter-1.4/plugins/specter_PGSQL.c	2008-01-07 00:13:54 +0000
@@ -73,7 +73,8 @@
 		+ strlen(GET_CE(ce,1)->u.string)
 		+ strlen(GET_CE(ce,2)->u.string)
 		+ strlen(GET_CE(ce,3)->u.string)
-		+ (GET_CE(ce,6)->u.value) ? 13 : 0;
+		+ ((GET_CE(ce,6)->u.value) ? 13 : 0)
+		+ strlen(GET_CE(ce,7)->u.string);
 	if ((initbuff = malloc(initlen)) == NULL) {
 		specter_log(SPECTER_FATAL, "Couldn't allocate data: %s.\n",
 				strerror(errno));

